src/test/func/memory/memory.cc

```diff
@@ -255,8 +255,8 @@ void test_external_pointer()
       if ((size_t)p4 != (size_t)p1 + size - 1)
       {
         std::cout << "size: " << size << " end(p4): " << p4 << " p1: " << p1
-                  << "  p1+size-1: " << (void*)((size_t)p1 + size - 1)
-                  << std::endl;
+                  << "  p1+size-1: "
+                  << (void*)(uintptr_t)((size_t)p1 + size - 1) << std::endl;
```

Comment:
The `(void*)` here are due to me being lazy and wanting to print the address in hex, and that is how pointers are printed.  We could fix this with `std::hex << `, or in this case actually using `pointer_offset`.
```suggestion
                  << pointer_offset(p1, size - 1) << std::endl;
```